_posts/2021-12-28-trino-2021-a-year-of-growth.md

```diff
@@ -0,0 +1,356 @@
+* Fixed incorrect results for queries that contain aggregations and `IN` and 
+`NOT IN` filters over varchar columns.
+* Fixed failure for queries with filters on `real` or `double` columns having 
+`+Infinity` or `-Infinity` values.
+* Implemented aggregation pushdown for `count`, `avg`, `min`, `max`, `sum`, 
+`count(DISTINCT)` and `approx_distinct`. It is enabled by default and can be 
+disabled using the configuration property `pinot.aggregation-pushdown.enabled` 
+or the catalog session property `aggregation_pushdown_enabled`.
+* Allowed https URLs in `pinot.controller-urls`. 
+
+### JDBC connectors
+
+* Added new configuration property `insert.non-transactional-insert.enabled` to 
+allow `INSERT` queries to write directly to the target table. This can improve 
+performance in some cases by sacrificing transactional behaviour. It can also be
+controlled using `non_transactional_insert` session property.
+* Added `write.batch-size` connector configuration property to control JDBC batch 
+size used during writes.
+* Improved metadata caching hit rate.
+
+## Runtime improvements
+
+* Improved coordinator CPU utilization.
+* Improved query performance by reducing CPU overhead of repartitioning data 
+across worker nodes.
+* Reduced graceful shutdown time for worker nodes.
+
+## Everything else
+
+* Added support for ARM64 in the Trino Docker image.
+* Added cancelling a Trino query execution when JDBC statement is closed.
+* Added `clear` command to the Trino CLI to clear the screen.
+* Improved tab completion for the Trino CLI.
+
+## Trino Summit
+
+In 2021 we also enjoyed a successful inaugural Trino Summit, hosted by 
+Starburst, with well over 500 attendees. There were wonderful talks
+given at this event from companies like Netflix, LinkedIn, EA, Robinhood, 
+Doordash, Tabular, and Stream Native. If you missed this event, we have the 
+[recordings and slides available](https://www.starburst.io/resources/trino-summit/).
+
+As a teaser, the event started with Commander Bun Bun playing guitar to AC/DC's,
+"Back In Black".
+
+<iframe src="https://www.youtube.com/embed/c_qUp0SGeKE"  
+width="800" height="500" frameborder="0" marginwidth="0" marginheight="0" 
+scrolling="no" style="border:1px solid #CCC; border-width:1px; 
+margin-bottom:5px; max-width: 100%;" allowfullscreen> 
+</iframe>
+
+## 2022 Roadmap: Project Tardigrade
+
+One of the interesting developments that came out of Trino Summit was a feature
+Trino co-creator, Martin, talked about in [the State of Trino presentation](https://www.starburst.io/resources/trino-summit/?wchannelid=2ug6mgs5ao&wmediaid=o264qw85dj).
+He discussed a prioritization in granular fault tolerance and performance in the
+core engine. While Trino has been proven to run batch analytics workloads at 
+scale, many have avoided long-running batch jobs in fear of a query failure. The
+fault-tolerance feature introduces a direction that the Trino project is taking 
```

Comment:
The fault-tolerance feature introduces a first step for the Trino project to gain first class support for long-running  batch queries at massive scale.


of making the Trino query also suitable for long-running